### ISSUE TITLE
TypeExtension.IsPlainType seems to misjudge the string property

### ISSUE THREAD

**Issue by fredliex** [mikependon/RepoDB#650]
For the case `new { a = "abc" }`, TypeExtension.IsPlainType currently returns false.
Should it return true?

`GetEnumerableClassProperties` was originally
```
internal static IEnumerable<ClassProperty> GetEnumerableClassProperties(this Type type) =>
    PropertyCache.Get(type).Where(classProperty => 
        StaticType.IEnumerable.IsAssignableFrom(classProperty.PropertyInfo.PropertyType));
```
Should it be changed to something like the following?
```
internal static IEnumerable<ClassProperty> GetEnumerableClassProperties(this Type type) =>
    PropertyCache.Get(type).Where(classProperty => 
    {
        var propType = classProperty.PropertyInfo.PropertyType;
        return
            propType != StaticType.String &&
            propType != StaticType.CharArray &&
            propType != StaticType.ByteArray &&
            StaticType.IEnumerable.IsAssignableFrom(propType);
    });
```

**Comment by mikependon** [comment]
@fredliex - I had tested this one and your report is valid. It seems you are correct as well, if we are to interpret the method name itself. Let me first explore the usage of this internally to avoid any effect on other use-cases. Will get back to you soon.

[EDIT]

```csharp
static class TypeExtensionEx
{
    /// <summary>
    /// Checks whether the current type is a plain class type.
    /// </summary>
    /// <param name="type">The current type.</param>
    /// <returns>Returns true if the current type is a plain class type.</returns>
    internal static bool IsPlainType(this Type type) =>
        (type.IsClassType() || type.IsAnonymousType()) &&
        IsQueryObjectType(type) != true &&
        type.IsDictionaryStringObject() == false &&
        GetEnumerableClassProperties(type).Any() != true;

    /// <summary>
    /// Checks whether the curren type is of type <see cref="QueryField"/> or <see cref="QueryGroup"/>.
    /// </summary>
    /// <param name="type">The curren type.</param>
    /// <returns>Returns true if the current type is of type <see cref="QueryField"/> or <see cref="QueryGroup"/>.</returns>
    internal static bool IsQueryObjectType(this Type type) =>
        type == typeof(QueryField) || type == typeof(QueryGroup);

    /// <summary>
    /// Gets the list of enumerable <see cref="ClassProperty"/> objects of the type.
    /// </summary>
    /// <param name="type">The current type.</param>
    /// <returns>The list of the enumerable <see cref="ClassProperty"/> objects.</returns>
    internal static IEnumerable<ClassProperty> GetEnumerableClassProperties(this Type type) =>
        PropertyCache.Get(type).Where(classProperty =>
            typeof(System.Collections.IEnumerable).IsAssignableFrom(classProperty.PropertyInfo.PropertyType));
}
```

At the Program.cs.

```csharp
static void Main(string[] args)
{
    var isPlainTypeForAnonymous = (new { Property = "ABC" }).GetType().IsPlainType();
    var isPlainTypeForClass = new Person().GetType().IsPlainType();
}
```

**Comment by mikependon** [comment]
@fredliex - hey, thank you for filing this issue and also for sleuthing around. I would like to record your recommendation, the fix you suggested here is the correct fix as I verified. With this, would you be able to make a PR for this one with your suggestions at the `GetEnumerableClassProperties()` method? Thanks

**Comment by fredliex** [comment]
I make PR #657 for this issue.

In addition.
I encountered a few issues related to date and string in RepoDb.IntegrationTests, which seems to be due to culture.
I will create an issue to describe it.

thank you.

**Comment by mikependon** [comment]
GH auto close the ticket, but this one is not yet closed until it is deployed, so I will just reopen it. Thanks for your PR. In relation to the culture issues on the Dates of the Integration Tests, yes, please do report it separately, so we can track it there. Thanks for being active.

**Comment by mikependon** [comment]
The fix is now available at [RepoDb v1.12.5-beta4](https://www.nuget.org/packages/RepoDb/1.12.5-beta4).

**Comment by mikependon** [comment]
Closing this ticket now.
